--- origsrc/paprefs-0.9.10/src/Makefile.am	2009-12-05 14:20:20.000000000 -0600
+++ src/paprefs-0.9.10/src/Makefile.am	2011-11-22 19:30:53.498134300 -0600
@@ -15,8 +15,6 @@
 # You should have received a copy of the GNU General Public License
 # along with paprefs. If not, see <http://www.gnu.org/licenses/>.
 
-# Needs fixing on some borked OS
-SHREXT=.so
 
 desktopdir = $(datadir)/applications
 gladedir = $(pkgdatadir)
@@ -31,7 +29,7 @@ paprefs_SOURCES=paprefs.cc
 
 paprefs_LDADD=$(AM_LDADD) $(GUILIBS_LIBS) $(LIBPULSE_LIBS)
 paprefs_CXXFLAGS=$(AM_CXXFLAGS) $(GUILIBS_CFLAGS) $(LIBPULSE_CFLAGS)
-paprefs_CXXFLAGS+=-DGLADE_FILE=\"$(gladedir)/paprefs.glade\" -DLOCALEDIR=\"$(localedir)\" -DMODLIBDIR=\""$(LIBPULSE_MODLIBDIR)/"\" -DSHREXT=\"$(SHREXT)\"
+paprefs_CXXFLAGS+=-DGLADE_FILE=\"$(gladedir)/paprefs.glade\" -DLOCALEDIR=\"$(localedir)\" -DMODLIBDIR=\""$(LIBPULSE_MODLIBDIR)/"\"
 
 EXTRA_DIST = $(glade_DATA) $(desktop_in_files)
 
--- origsrc/paprefs-0.9.10/src/paprefs.cc	2011-03-20 09:01:13.000000000 -0500
+++ src/paprefs-0.9.10/src/paprefs.cc	2011-11-22 19:30:15.953986900 -0600
@@ -33,6 +33,7 @@
 
 #include <pulse/version.h>
 
+#define SHREXT "."G_MODULE_SUFFIX
 #define PA_GCONF_ROOT "/system/pulseaudio"
 #define PA_GCONF_PATH_MODULES PA_GCONF_ROOT"/modules"
 
